app/controllers/submit.js

```diff
@@ -333,8 +338,38 @@ export default Ember.Controller.extend(BasicsValidations, NodeActionsMixin, Tagg
             this.send('next', this.get('_names.0'));
         },
         resetFileUpload() {
-            this.send('changeState', State.START);
-            this.set('fileAndNodeLocked', false);
+            var promisesArray = [];
+            var filePromises = [];
+            this.get('filesUploadedForPreprint').forEach((file) => {
+                filePromises.push(this.get('fileManager').deleteFile(file).then(() => {
+                    this.get('toast').info('File removed!');
+                }).catch(() => {
+                    this.get('toast').error('Could not remove uploaded file');
+                }));
+            });
+
+            Ember.RSVP.all(filePromises).then(() => {
```

Comment:
This seems to wrap the success or failure states of many different operations into one. Since all fails fast, it's possible that we could end up in a state where some files had been deleted, but not others, and we had not correctly given this page an accurate picture of reality at the end of these operations.

Would it be more appropriate to use `RSVP.allSettled` here, and check the failure state? (note the return values look a bit different for this method, see [the docs](http://emberjs.com/api/classes/RSVP.html#method_allSettled))

I could reasonably consider this to be something for a followup PR, but it is an error case to be aware of.